### PR TITLE
Bugfix/metadata ioerror

### PR DESCRIPTION
**Pull request by sharkykh**
"Fix" #4499 & #4500 
Permission errors are user errors, log them as warnings.

Also use errno package instead of comparing to integers.

### CODE REVIEW

File: medusa/helpers/__init__.py [hunk 1]
```diff
@@ -346,7 +346,7 @@ def hardlink_file(src_file, dest_file):
         link(src_file, dest_file)
         fix_set_group_id(dest_file)
     except OSError as msg:
-        if hasattr(msg, 'errno') and msg.errno == 17:
+        if hasattr(msg, 'errno') and msg.errno == errno.EXIST:
```
Comment (labrys) [inline medusa/helpers/__init__.py]: @sharkykh generally you don't LBYL in python.  `OSError` should always have an `errno` so the `hasattr` call is extraneous.

Comment (sharkykh) [reply medusa/helpers/__init__.py]: _(Ignoring the fact I didn't write that code)_
I couldn't find any documentation that clearly states OSError / IOError always have an `errno` attr.
Any chance you have a source for that claim?

Comment (labrys) [reply medusa/helpers/__init__.py]: @sharkykh From the Python 2 docs for [Built-in Exceptions](https://docs.python.org/2/library/exceptions.html):

#### For `OSError`:

> *exception* __OSError__
This exception is derived from `EnvironmentError`. It is raised when a function returns a system-related error (not for illegal argument types or other incidental errors). The `errno` attribute is a numeric error code from `errno`, and the `strerror` attribute is the corresponding string, as would be printed by the C function `perror()`. See the module `errno`, which contains names for the error codes defined by the underlying operating system.

#### For `IOError`:
> *exception* __IOError__
Raised when an I/O operation (such as a print statement, the built-in `open()` function or a method of a file object) fails for an I/O-related reason, e.g., “file not found” or “disk full”.
> This class is derived from `EnvironmentError`. See the discussion above for more information on exception instance attributes.

#### For the relevant `EnvironmentError` mentioned in both of the above:
> *exception* __EnvironmentError__
The base class for exceptions that can occur outside the Python system: `IOError`, `OSError`. When exceptions of this type are created with a 2-tuple, the first item is available on the instance’s `errno` attribute (it is assumed to be an error number), and the second item is available on the `strerror` attribute (it is usually the associated error message). The tuple itself is also available on the `args` attribute.

## TLDR:
> When exceptions of this type are created with a 2-tuple, the first item is available on the instance’s `errno` attribute

Comment (labrys) [reply medusa/helpers/__init__.py]: @sharkykh I know you didn't write the line, but I'm a firm believer that if you're editing code then you should take the time to understand and review the lines your modifying.

Comment (sharkykh) [reply medusa/helpers/__init__.py]: Thanks.

> if you're editing code then you should take the time to understand and review the lines your modifying.

That is precisely what I did (tried to do).
I tried to find info about it and somehow completely missed it (was probably tired).

Comment (labrys) [reply medusa/helpers/__init__.py]: Just to clarify, since its difficult to express tone in text, my comment wasn't meant to imply that you don't but as a clarification of the reason I tagged you specifically since you were the most recent editor of the lines.  I know from personal I experience I miss stuff like this all the time.  You tend to get code-blind when looking at it for so long 😁
